src/main/java/org/broadinstitute/hellbender/tools/pon/coverage/pca/HDF5PCACoveragePoNCreationUtils.java

```diff
@@ -350,150 +249,151 @@ public double visit(final int row, final int column, final double value) {
-        final Map<String, Double> columnsToDrop = new LinkedHashMap<>((int) (extremeColumnMedianCountPercentileThreshold * 4.0 / 100.0));
-        for (int i = 0; i < columnMedians.length; i++) {
-            if (columnMedians[i] >= bottomExtremeThreshold && columnMedians[i] <= topExtremeThreshold) {
-                columnsToKeep.add(columnNames.get(i));
-            } else {
-                columnsToDrop.put(columnNames.get(i), columnMedians[i]);
-            }
-        }
+    /**
+     * Calculate the target factors (median coverage per target).
+     *
+     * @param readCounts    the input read counts
+     * @return              never {@code null}, with as many elements as targets in {@code readCounts}
+     */
+    private static double[] calculateTargetFactors(final ReadCountCollection readCounts) {
+        Utils.nonNull(readCounts, "ReadCountCollection cannot be null.");
+        return MatrixSummaryUtils.getRowMedians(readCounts.counts());
+    }
 
-        // log and drop columns if it applies
-        if (columnsToKeep.isEmpty()) {
-            throw new UserException.BadInput("No column count left after applying the extreme counts outlier filter");
-        } else if (columnsToKeep.size() == columnNames.size()) {
-            logger.info(String.format("No column dropped due to extreme counts outside [%.10f, %.10f]", bottomExtremeThreshold, topExtremeThreshold));
+    /**
+     * Calculates the absolute number of zeroes tolerated given the user requested percentage
+     * and the number of counts involved.
+     *
+     * @param totalCounts   a non-negative number of counts to check
+     * @param percentage    a value in [0, 100]
+     * @return              a value between 0 and {@code totalCounts}
+     */
+    private static int calculateMaximumZerosCount(final int totalCounts, final double percentage) {
+        ParamUtils.isPositiveOrZero(totalCounts, "Total counts must be non-negative.");
+        ParamUtils.inRange(percentage, 0, 100, "Percentage must be in [0, 100].");
+        return (int) Math.ceil(totalCounts * percentage / 100.0);
+    }
+
+    /**
+     * Remove columns that have too many counts equal to 0.
+     * <p>
+     *     It will return a copy of the input read-count collection with such columns dropped.
+     * </p>
+     *
+     * @param readCounts the input read counts.
+     * @param maximumColumnZeros maximum number of counts equal to 0. per column tolerated.
+     * @return never {@code null}. It might be a reference to the input read-counts if there is
+     *   is no column to be dropped.
+     */
+    @VisibleForTesting
+    static ReadCountCollection removeColumnsWithTooManyZeros(final ReadCountCollection readCounts, final int maximumColumnZeros, final Logger logger) {
+        final RealMatrix counts = readCounts.counts();
+
+        final Set<String> columnsToKeep = IntStream.range(0, counts.getColumnDimension()).boxed()
```

Comment:
Instead of `.boxed() . . . map()` how about skip the `boxed()` and use `mapToObj()`?